src/mpbuild/build.py

```diff
@@ -26,14 +26,14 @@
 
 
 def build_board(
-    port: str, board: str, variant: Optional[str] = None, idf: Optional[str] = None
+    port: str, board: str, variant: Optional[str] = None, extra_args: Optional[List[str]] = [], idf: Optional[str] = None
 ) -> None:
     if port not in _build_containers.keys():
         print(f"Sorry, builds are not supported for the {port} port at this time")
         # TODO(mst) Should raise an exception and abort with an error code
         return
 
-    if port != "esp32" and idf:
+    if port != "esp32" and idf != IDF_DEFAULT:
```

Comment:
I think that logic is incorrect; it should be testing for `None`. (If the user used a different port and supplied a non-default IDF an error should still be raised)

Reply:
The problem is, I wanted to show what the default IDF value is in help (see printout above). Once it's the default value though, it's not None here anymore

Reply:
Ah, I see, tricky...

Reply:
I figured the new help message would assist is people only using when needed... and if they set it anyway well it just doesn't do anything :-D

Reply:
Could test if it equals the default (from `cli.py`) and then set it to `None` if it does. Then `build.py` will re-set it to the default if the esp32 port is selected - or generate an error if it is. 

It's an ugly workaround though!

Reply:
Yeah, I may be over-thinking this as an error case worth handling...

Reply:
However resetting from default to None then checking for None results in exactly the same behavior as the code I've got here (I thought of doing the same thing)!

If the user sets anything that isn't default, the error will be shown. If the user manually sets the same value as the default, there will be no error shown.